content/shmem_for.tex

```diff
@@ -0,0 +1,45 @@
+\apisummary{
+  Atomically perform a fetching bitwise inclusive or operation on a
+  remote data object.
+}
+
+\begin{apidefinition}
+
+\begin{C11synopsis}
+TYPE shmem_for(TYPE *dest, TYPE value, int pe);
```

Comment:
I'm leaning toward overhauling the AMOs with the naming scheme I sketched in [this comment](#issuecomment-273598859). I think the explicit nature of the names is generally an improvement (despite their length) over the short-form names we currently have (which don't explicitly mention atomicity) because they clearly convey the intent of the function call.

I think this probably makes sense to issue as two PRs: one for bitwise AMOs with the "new style" names and one to rename existing AMOs (deprecate "old style" and add "new style" names).

Thoughts?